### PR TITLE
Fix core editor store available in classic

### PR DESCRIPTION
**Pull request by chihsuan**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

p1696248781817279-slack-C7KF0SHHQ

In the 8.2.0-beta, the core-editor Redux store is available in the classic product editor but it’s non-functional. This doesn't cause bugs for users but affects some plugins because they rely on the logic of whether the wp-editor store exists.

Since we only use it in customize your store area, this PR removes `wp-editor` from the dependencies array if we're not on a customize store page.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to /wp-admin/post-new.php?post_type=product
2. Open dev console
3. Enter `wp.data.select('core/editor')`
4. Should return `undefined`.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Internal/Admin/WCAdminAssets.php [hunk 1]
```diff
@@ -305,6 +305,16 @@ public function register_scripts() {
 					$script_assets['dependencies'] = array_diff( $script_assets['dependencies'], array( 'wp-router' ) );
 				}
 
+				// Remove wp-editor dependency if we're not on a customize store page since we don't use wp-editor in other pages.
+				$is_customize_store_page = (
+					PageController::is_admin_page() &&
+					isset( $_GET['path'] ) &&
+					str_starts_with( wc_clean( wp_unslash( $_GET['path'] ) ), '/customize-store' )
+				);
+				if ( ! $is_customize_store_page ) {
```
Comment (louwie17) [inline plugins/woocommerce/src/Internal/Admin/WCAdminAssets.php]: I would also add ` && WC_ADMIN_APP === $script`, given the customizer logic only resides in the WooCommerce app. That way all other packages that depend on it, still have it included.

**Review by louwie17 — CHANGES_REQUESTED** [review]
Requested one change for the logic, and otherwise I think its good to merge.

Although I would also prefer to flip this logic and have the `DependencyExtraction` webpack plugin within `woocommerce-admin` webpack plugin specifically exclude `@wordpress/editor`: https://github.com/woocommerce/woocommerce/blob/f73b8a80f6d47a871bed01d8d288cf29e0ab5bfe/plugins/woocommerce-admin/webpack.config.js#L225-L226
```
requestToHandle( request ) {
  if ( request.startsWith( '@wordpress/editor' ) ) {
	return null;
  }
},
```
Then you can specifically enqueue it only on the StoreCustomizer page.
In theory it would be the exact same logic, but the exclusion of the @wordpress/editor is more visible this way.
Although I noticed the above doesn't work out of the box with the extraction plugin and there is actually a minor bug in that plugin that doesn't allow us to return `null` to skip it. So we did have to make a minor change there as well.
This maybe the better route for now.

File: plugins/woocommerce/src/Internal/Admin/WCAdminAssets.php (hunk 1 above)
Comment (chihsuan) [reply plugins/woocommerce/src/Internal/Admin/WCAdminAssets.php]: Sounds good! Added in 9b8ea937b75c12acc3c2d962163905f8a52940bb

**Comment by chihsuan** [comment]
> Then you can specifically enqueue it only on the StoreCustomizer page.
> In theory it would be the exact same logic, but the exclusion of the @wordpress/editor is more visible this way.
> Although I noticed the above doesn't work out of the box with the extraction plugin and there is actually a minor bug in that plugin that doesn't allow us to return `null` to skip it. So we did have to make a minor change there as well.
> This maybe the better route for now.

Thanks for the review! @louwie17 🙏  Yeah, that would be better. I've added ` && WC_ADMIN_APP === $script`.

**Review by louwie17 — APPROVED** [review]
Tested well and thanks for the code update, this looks great!! Appreciate that you addressed this :)

**Pull request merged**